tests/test_read_parsers.py

```diff
@@ -483,5 +483,204 @@ def test_clean_seq():
+    assert dist[2] == 42
+
+
+def test_read_cleaning_trim_functions_lowercase():
+    infile = utils.get_test_data('valid-read-testing.fq')
+
+    # read this in using "approved good" behavior w/cleaned_seq
+    x = Counttable(8, int(1e6), 4)
+    for read in ReadParser(infile):
+        x.consume(read.cleaned_seq)       # consume cleaned_seq
+
+    s = "caggcgcccaccaccgtgccctccaacctgatggt"
+    _, where = x.trim_on_abundance(s, 1)
+    assert where == 35                    # in future, should be '0'
+
+    _, where = x.trim_below_abundance(s, 2)
+    assert where == 35                    # in future, should be ?? @CTB
+
+    posns = x.find_spectral_error_positions(s, 1)
+    assert posns == []                    # in future, should do same
+
+
+def test_read_cleaning_trim_functions_N():
+    infile = utils.get_test_data('valid-read-testing.fq')
+
+    # read this in using "approved good" behavior w/cleaned_seq
+    x = Counttable(8, int(1e6), 4)
+    for read in ReadParser(infile):
+        x.consume(read.cleaned_seq)       # consume cleaned_seq
+
+    s = "ACTGGGCGTAGNCGGTGTCCTCATCGGCACCAGC"
+    _, where = x.trim_on_abundance(s, 1)
+    assert where == 0                     # in future, should be ??
+
+    _, where = x.trim_below_abundance(s, 2)
+    assert where == 0                     # in future, should be ??
+
+    with pytest.raises(ValueError):
+        posns = x.find_spectral_error_positions(s, 1)
+    # assert posns == []                    # in future, should return []
+
+
+def test_read_cleaning_trim_functions_bad_dna():
+    infile = utils.get_test_data('valid-read-testing.fq')
+
+    # read this in using "approved good" behavior w/cleaned_seq
+    x = Counttable(8, int(1e6), 4)
+    for read in ReadParser(infile):
+        x.consume(read.cleaned_seq)       # consume cleaned_seq
+
+    s = "CCGGCGTGGTTZZYAGGTCACTGAGCTTCATGTC"
+    _, where = x.trim_on_abundance(s, 1)
+    assert where == 0                     # in future, should be ??
+
+    _, where = x.trim_below_abundance(s, 2)
+    assert where == 0                     # in future, should be ??
+
+    with pytest.raises(ValueError):
+        posns = x.find_spectral_error_positions(s, 1)
+    # assert posns == []                    # in future, should return [11]
+
+
+def test_read_cleaning_output_partitions():
+    infile = utils.get_test_data('valid-read-testing.fq')
+    savepath = utils.get_temp_filename('foo')
+
+    # read this in using "approved good" behavior w/cleaned_seq
+    x = Nodegraph(8, int(1e6), 4)
+    for read in ReadParser(infile):
+        x.consume(read.cleaned_seq)       # consume cleaned_seq
+
+    kmer = 'caggcgcc'.upper()
+    x.add_tag(kmer)
+    x.set_partition_id(kmer, 1)
+
+    kmer = 'ACTGGGCG'
+    x.add_tag(kmer)
+    x.set_partition_id(kmer, 2)
+    
+    kmer = 'CCGGCGTG'
+    x.add_tag(kmer)
+    x.set_partition_id(kmer, 3)
+    
+    x.output_partitions(infile, savepath)
+
+    read_names = [ read.name for read in ReadParser(savepath) ]
```

Comment:
The `pep8` :police_car: doesn't like the extra white space (hence the travis failure)